### PR TITLE
Create ServerAPI helper method for enrichment endpoints #814

### PR DESCRIPTION
**Pull request by CaitlinOCallaghan**
Addresses portion of #814

'geneQuery' method returns 'enrich' method for validation
if/else case removed

### CODE REVIEW

File: src/client/services/server-api/index.js [hunk 1]
```diff
@@ -32,19 +32,19 @@ const ServerAPI = {
     return fetch(`/pc-client/querySearch?${qs.stringify(queryClone)}`, defaultFetchOpts).then(res => res.json());
   },
 
+  enrich(query, type){
```
Comment (jvwong) [inline src/client/services/server-api/index.js]: `enrichmentAPI`

File: src/client/services/server-api/index.js [hunk 2]
```diff
@@ -32,19 +32,19 @@ const ServerAPI = {
...
+    return fetch(`/api/${type}`, {
+      method:'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body:JSON.stringify(query)
+    }).then(res => res.json());
+  },
```
Comment (jvwong) [inline src/client/services/server-api/index.js]: Let's make a habit of putting a `catch` after these.

**Comment by CaitlinOCallaghan** [comment]
Since this is for the enrichment API, can we send the error code 400 and message for the catch? Please let me know if I should change the contents of the catch.

**Comment by jvwong** [comment]
It's unlikely to be a Bad Request (400). You can try [the solutions from the docs](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Checking_that_the_fetch_was_successful) if you want.

File: src/client/services/server-api/index.js [hunk 3]
```diff
@@ -42,9 +42,7 @@ const ServerAPI = {
       body:JSON.stringify(query)
     })
     .then(res => res.json())
-    .catch((err) => {
-      res.status(400).send(err.message);
-  },
+    .catch((err) => { console.error('Error:', err); },
```
Comment (jvwong) [inline src/client/services/server-api/index.js]: Is this missing closing brackets?

**Pull request merged**